pipelines/batch/src/main/java/org/openmrs/analytics/JdbcFetchHapi.java

```diff
@@ -160,4 +165,34 @@ List<QueryParameterDescriptor> generateQueryParameters(
 
     return queryParameterList;
   }
+
+  /**
+   * Searches for the total number of resources for each resource type
+   *
+   * @param resourceList the resource types to be processed
+   * @param since the time from which the records need to be fetched
+   * @return a Map storing the counts of each resource type
+   */
+  public Map<String, Integer> searchResourceCounts(String resourceList, String since)
+      throws SQLException {
+    HashSet<String> resourceTypes = new HashSet<String>(Arrays.asList(resourceList.split(",")));
+    HashMap<String, Integer> hashMap = new HashMap<String, Integer>();
```

Comment:
these can be "Program to Interface" e.g.
Set<String> resourceTypes = new HashSet<String>(Arrays.asList(resourceList.split(",")));
Map<String, Integer> hashMap = new HashMap<>(); and notice we dont need to provide the types in HashMap<>

Reply:
also, can we have a better name instead of hashMap

Reply:
Done